tests/realm.cpp

```diff
@@ -1284,6 +1284,25 @@ TEST_CASE("SharedRealm: dynamic schema mode doesn't invalidate object schema poi
     REQUIRE(object_schema == &*r2->schema().find("object"));
 }
 
+TEST_CASE("SharedRealm: declaring an object as embedded results in creating an embedded table") {
```

Comment:
The existing tests for validating that we created the correct things in core from a given schema are in migrations.cpp. Adding sections for is_embedded to each of the existing test cases there will probably reveal some more scenarios that need to be handled.

Reply:
Ok, I will look into that.